service/routers/decks.js

```diff
@@ -0,0 +1,184 @@
+  }
+  
+}
+
+const deleteDeck = async (req, res) => {
+  if(req.user.role == "Admin" || req.user.role == "SuperUser" || req.user.userId == req.params.id) {
+    const { userId, role } = req.user
+    const deckId = req.params.deckId
+    try {
+      const user = await User.findById(req.params.id)
+      const removedDeck = user.decks.id(deckId).remove()
+      console.log(removedDeck)
+      user.save()
+      res.sendStatus(204)
+    } catch (err) {
+      console.log(`${deleteDeck.name}: ${err}`)
+      res.sendStatus(500)
+    }
+  } else {
+    res.status(404).send('you are not authorized to access this route')
+  }
+}
+
+const updateDeck = async (req, res) => {
+  if(req.user.role == "Admin" || req.user.role == "SuperUser" || req.user.userId == req.params.id) {
+    const { userId, role } = req.user
+    const deckId = req.params.deckId
+    const newDeck = req.body
+    try {
+      const user = await User.findById(req.params.id)
+      const deck = user.decks.id(deckId)
+      deck.name = newDeck.name
+      await user.save()
+      res.sendStatus(204)
+    } catch (err) {
+      console.log(`${updateDeck.name}: ${err}`)
+      res.sendStatus(500)
+    }
+  } else {
+    res.status(404).send('you are not authorized to access this route')
+  }
+}
+
+const updateCard = async (req, res) => {
+  if(req.user.role == "Admin" || req.user.role == "SuperUser" || req.user.userId == req.params.id) {
+    const { userId, role } = req.user
+    const deckId = req.params.deckId
+    const cardId = req.params.cardId
+    const newCard = req.body
+    try {
+      const user = await User.findById(req.params.id)
+      const deck = user.decks.id(deckId)
+      const card = deck.cards.id(cardId)
+      card.frontImage = newCard.frontImage
+      card.frontText = newCard.frontText
+      card.backImage = newCard.backImage
+      card.backText = newCard.backText
+      await user.save()
+      res.sendStatus(204)
+    } catch (err) {
+      console.log(`${createCard.name}: ${err}`)
+      res.sendStatus(500)
+    }
+  } else {
+    res.status(404).send('you are not authorized to access this route')
+  }
+}
+
+const deleteCard = async (req, res) => {
+  if(req.user.role == "Admin" || req.user.role == "SuperUser" || req.user.userId == req.params.id) {
+    const { userId, role } = req.user
+    const deckId = req.params.deckId
+    const cardId = req.params.cardId
+    try {
+      const user = await User.findById(req.params.id)
+      const deck = user.decks.id(deckId)
+      const removedCard = deck.cards.id(cardId).remove()
+      console.log(removedCard)
+      user.save()
+      res.sendStatus(204)
+    } catch (err) {
+      console.log(`${deleteCard.name}: ${err}`)
+      res.sendStatus(500)
+    }
+  } else {
+    res.status(404).send('you are not authorized to access this route')
+  }
+}
+
+decksRouter.get('/:id', getDecks)
+decksRouter.post('/:id', body('name').not().isEmpty(), createDeck)
```

Comment:
If you need an id for a user in this case, you should be passing that in the body of the request.